client/src/hooks/board/useCanvasEventListeners.tsx

```diff
@@ -118,37 +146,41 @@ const useCanvasEventHandlers = (
     };
 
     const handleTextEditingExited = (e: any) => {
-      if (e.target._id !== state.observedTextId || e.target.text === state.observedTextContent) {
+      if (e.target.text === state.observedTextContent) {
+        return;
+      }
+      if (!socket) {
         return;
       }
 
       const targetJSON = e.target.toJSON(["_id"]);
       const clonedJSON = { ...targetJSON, text: state.observedTextContent };
 
-      const command = new ModifyCommand(canvas, clonedJSON, targetJSON, handleStyleChange);
+      const updateObjectData: UpdateObjectData = {
+        object: targetJSON,
+      };
+      socketEmitUpdateObject(socket, updateObjectData);
+
+      const command = new ModifyCommand(canvas, clonedJSON, targetJSON, targetJSON._id, handleStyleChange);
       saveCommand(command);
 
       dispatch({ type: "SET_OBSERVED_TEXT_CONTENT", payload: e.target.text });
     };
 
-    // Register event listeners
-    canvas.on("path:created", handlePathCreated);
-    canvas.on("object:modified", handleObjectModified);
-    canvas.on("erasing:end", handleEraserAdded);
-    canvas.on("text:editing:entered", handleTextEditingEntered);
-    canvas.on("text:editing:exited", handleTextEditingExited);
-    canvas.on("selection:created", handleMultipleSelections);
-    canvas.on("selection:updated", handleMultipleSelections);
+    const listeners = [
+      { event: "path:created", handler: handlePathCreated },
+      { event: "object:modified", handler: handleObjectModified },
+      { event: "erasing:end", handler: handleEraserAdded },
+      { event: "text:editing:entered", handler: handleTextEditingEntered },
+      { event: "text:editing:exited", handler: handleTextEditingExited },
+      { event: "selection:created", handler: handleMultipleSelections },
+      { event: "selection:updated", handler: handleMultipleSelections },
```

Comment:
I think we should try and implement this functionality (because it's quite useful), but not on this branch. Perhaps it could be a part of another task?
Another thing we should consider is: should we persist these very frequent changes? It feels like the easiest way to kill our database. I think these frequent throttled changes should be broadcasted to other users connected to the board, but shouldn't be persisted.
That's why I think this functionality requires a bit more things to do. Thus, it should be a separate task

Reply:
Okay i created it